src/vos/vos_dtx.c

```diff
@@ -2089,13 +2089,21 @@ vos_dtx_abort(daos_handle_t coh, struct dtx_id *dti, daos_epoch_t epoch)
 	d_iov_set(&kiov, dti, sizeof(*dti));
 	d_iov_set(&riov, NULL, 0);
 	rc = dbtree_lookup(cont->vc_dtx_active_hdl, &kiov, &riov);
+	if (rc == -DER_NONEXIST) {
+		rc = dbtree_lookup(cont->vc_dtx_committed_hdl, &kiov, &riov);
+		if (rc == 0) {
+			D_ERROR("NOT allow to abort a committed DTX (1) "DF_DTI"\n", DP_DTI(dti));
+			D_GOTO(out, rc = -DER_NO_PERM);
```

Comment:
What's the difference of returning NONEXIST or NO_PERM error? Will caller treat them differently?

I mean, given that the 'vc_dtx_committed_hdl' could be aggregated, we actually can't tell if a transaction is committed or not by lookup the 'vc_dtx_committed_hdl', right? Which means NONEXIST could be returned for a committed transaction as well.

Reply:
NONEXIST means cannot find related DTX entry currently, it my because the DTX entry has never been there, also may be removed by aggregation. On some degree, we can consider that: dtx_abort is one kind of DTX remove operation, dtx_aggregation is another kind of remove operation. If one DTX entry has already been removed, it cannot be removed again. But if the DTX entry is there, then related remove operation needs to have proper permission and return more clear errno to the caller if we exactly know the failure reason. As for how the caller will use the returned errno, that depends on the caller itself, maybe case by case.

Reply:
Ok, so it looks to me there is no difference of returning NONEXIST or NO_PERM. My point is that the caller has to treat these two errors in the same way, because it can't tell if a handle is committed or not by the error code. (committed handle could return NONEXIST or NO_PERM)

But I'm ok with landing this patch as-is.